### ISSUE TITLE
Plugin `create_lr_scheduler` is not used

### ISSUE THREAD

**Issue by alexdremov** [axolotl-ai-cloud/axolotl#2576]
### Please check that this issue hasn't been reported before.

- [x] I searched previous [Bug Reports](https://github.com/axolotl-ai-cloud/axolotl/labels/bug) didn't find any similar reports.

### Expected Behavior

Use `create_lr_scheduler` to create custom lr schedulers

### Current behaviour

Ignored and never called

### Steps to reproduce

Define `create_lr_scheduler` in plugin, check if it is called

### Config yaml

```yaml

```

### Possible solution

Use `create_lr_scheduler` in trainer builder if returned value is not None

### Which Operating Systems are you using?

- [x] Linux
- [x] macOS
- [x] Windows

### Python Version

3.11

### axolotl branch-commit

main

### Acknowledgements

- [x] My issue title is concise, descriptive, and in title casing.
- [x] I have searched the existing issues to make sure this bug has not been reported yet.
- [x] I am using the latest version of axolotl.
- [x] I have provided enough information for the maintainers to reproduce and diagnose the issue.

**Comment by NanoCode012** [comment]
Hey, this is true. Seems like we have two hooks not implemented `create_optimizer` and `create_lr_scheduler`. Would you be interested in making a PR for it?

**Comment by alexdremov** [comment]
> Hey, this is true. Seems like we have two hooks not implemented `create_optimizer` and `create_lr_scheduler`. Would you be interested in making a PR for it?

Sure, just was planning to make a fork :)

**Comment by alexdremov** [comment]
> Hey, this is true. Seems like we have two hooks not implemented `create_optimizer` and `create_lr_scheduler`. Would you be interested in making a PR for it?

Here it comes https://github.com/axolotl-ai-cloud/axolotl/pull/2584
